index.js

```diff
@@ -282,14 +292,18 @@ exports.stringify = (object, options) => {
 
 	const formatter = encoderForArrayFormat(options);
 
-	const objectCopy = Object.assign({}, object);
-	if (options.skipNull) {
-		for (const key of Object.keys(objectCopy)) {
-			if (objectCopy[key] === undefined || objectCopy[key] === null) {
-				delete objectCopy[key];
+	const objectCopy = Object
+		.keys(object)
+		.reduce((objectCopy, key) => {
```

Comment:
Don't do unrelated changes. I prefer this to stay a `for-of` loop instead of reduce, for readability.

Reply:
I didn't refactor, rather a related change

```javascript
if (options.skipNull) {
  for (const key of Object.keys(objectCopy)) {
    if (objectCopy[key] === undefined || objectCopy[key] === null) {
      delete objectCopy[key];
    }
  }
}
```

the same can be achieved with 

```javascript
if (options.skipNull || options. skipEmptyString) {
  for (const key of Object.keys(objectCopy)) {
    if (objectCopy[key] === undefined || objectCopy[key] === null || objectCopy[key] === '') {
      delete objectCopy[key];
    }
  }
}
```
suppose in future if we introduce other filters the list will go on...

is there any better way than this?

Reply:
> is there any better way than this?

Do the same as now, but instead of reduce, use `for-of` loop, meaning no `Object.asssign`. You could maybe move the filtering logic here into a helper function (in this scope, not top-level).

Reply:
hmm ok, if I remove the reduce this is how the code is (existing) will this work? sorry if I am not getting your point

```javascript
const objectCopy = Object.assign({}, object);

for (const key of Object.keys(object)) {
    if ((options.skipNull && isNullOrUndefined(object[key])) ||
        (options.skipEmptyString && object[key] === '')) {
        delete objectCopy[key];
    }
}
```

this is another approach

```javascript
const shouldFilter = key => (
    (options.skipNull && isNullOrUndefined(object[key])) ||
    (options.skipEmptyString && object[key] === '')
);

const formatter = encoderForArrayFormat(options);

const objectCopy = {};

for (const key of Object.keys(object)) {
    if (!shouldFilter(key)) {
        objectCopy[key] = object[key];
    }
}
```

we can add as many filters we need in future

your thoughts?

Reply:
^ The last code block here is what I had in mind. 👌 Sorry for being unclear.